lib/jsdom/browser/parser/html.js

```diff
@@ -16,6 +16,29 @@ const OpenElementStack = require("parse5/lib/parser/open-element-stack");
 const OpenElementStackOriginalPop = OpenElementStack.prototype.pop;
 const OpenElementStackOriginalPush = OpenElementStack.prototype.push;
 
+const adapterStack = [];
+
+// Horrible monkey-patch to implement https://github.com/inikulin/parse5/issues/237
+OpenElementStack.prototype.push = function (...args) {
+  OpenElementStackOriginalPush.apply(this, args);
+  adapterStack[adapterStack.length - 1]._currentElement = this.current;
```

Comment:
This should have a check for `adapterStack` not empty, just in case parse5 is used elsewhere not in jsdom.